tests/lib/rules/no-unneeded-ternary.js

```diff
@@ -167,6 +168,25 @@ ruleTester.run("no-unneeded-ternary", rule, {
             }]
         },
         {
+            code: `
+                var value = 'a'
+                var canSet = true
+                var result = value ? value : canSet ? 'unset' : 'can not set'
+            `,
+            output: `
+                var value = 'a'
+                var canSet = true
+                var result = value || (canSet ? 'unset' : 'can not set')
+            `,
+            options: [{ defaultAssignment: false }],
+            errors: [{
+                message: "Unnecessary use of conditional expression for default assignment.",
+                type: "ConditionalExpression",
+                line: 4,
+                column: 38
+            }]
```

Comment:
Can you add a test where `node.alternate` is already parenthesized? For example:

```js
foo ? foo : (bar ? baz : qux)
```

This should get fixed to

```js
foo || (bar ? baz : qux)
```

At the moment, I think this will result in `bar ? baz : qux` getting parenthesized twice, which isn't correct.